src/core/BUILD

```diff
@@ -2252,6 +2268,7 @@ grpc_cc_library(
         "default_event_engine_factory",
         "event_engine_trace",
         "no_destruct",
+        "thready_event_engine",
```

Comment:
We might save some compile time & bloat in the majority of builds if the `thready_event_engine` code had `GRPC_MAXIMIZE_THREADINESS` macro guards around everything.

Reply:
Bloat is probably not an issue - if these symbols are unused (and they will be) they won't be compiled in.
The compile time improvements would be marginal.
The win here though is it means we can run the relevant ThreadyEventEngine tests in all configurations (to say, catch leaks in the implementation) even though its main use is only in one build configuration.

Reply:
That's a good point. Looks good as is.